hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockDataStreamOutput.java

```diff
@@ -308,11 +335,10 @@ public void writeOnRetry(long len) throws IOException {
     int count = 0;
     while (len > 0) {
       final StreamBuffer buf = bufferList.get(count);
-      final long writeLen = Math.min(buf.length(), len);
+      final long writeLen = Math.min(buf.getBuffer().position(), len);
       final ByteBuffer duplicated = buf.duplicate();
-      if (writeLen != buf.length()) {
-        duplicated.limit(Math.toIntExact(len));
-      }
+      duplicated.position(0);
```

Comment:
will this position be always be set to 0?? The starting offset should be deduced rom last ack'd length in case of retry.

Reply:
We are writing a new chunk here everytime picking each entry from the bufferList, hence the position will be 0.  Same logic as [this](https://github.com/apache/ozone/blob/99ddf5a80b4bec59f165b113154ae61416b4fac8/hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java#L506)